cdptools/natural_language_analyzers/entity_analyzer.py

```diff
@@ -1,14 +1,352 @@
+    event_time: datetime
+        The timestamp that the event occurred on
+
+    Returns
+    -------
+    annotations: List[EntityAnnotation]
+        An empty list.
+    """
+    return []
+
+
+def _annotate_person_entities(
+    entities: List[spacy.tokens.Span],
+    doc: spacy.tokens.Doc,
+    event_time: datetime
+) -> List[EntityAnnotation]:
+    """
+    Produce annotations for person entities.
+
+    This filters out entities for ones containing spaces, using that as a proxy
+    to indicate first and last name.
+
+    Parameters
+    ----------
+    entities: List[spacy.tokens.Span]
+        A list of spacy.tokens.Span objects representing entities
+    doc: spacy.tokens.Doc
+        The spacy doc the entities are derived from
+    event_time: datetime
+        The timestamp that the event occurred on
+
+    Returns
+    -------
+    annotations: List[EntityAnnotation]
+        A list of absolute date annotations.
+    """
+    return [
+        _span_to_annotation(e, "Entity[Person]", e.text)
+        for e in entities
+        if " " in e.text
+    ]
+
+
+def _annotate_date_entities(
+    entities: List[spacy.tokens.Span],
+    doc: spacy.tokens.Doc,
+    event_time: datetime
+) -> List[EntityAnnotation]:
+    """
+    Parse exact dates and transform relative dates into exact dates based on date and
+    time the event ocurred.
+
+    Parameters
+    ----------
+    entities: List[spacy.tokens.Span]
+        A list of spacy.tokens.Span objects representing entities
+    doc: spacy.tokens.Doc
+        The spacy doc the entities are derived from
+    event_time: datetime
+        The timestamp that the event occurred on
+
+    Returns
+    -------
+    annotations: List[EntityAnnotation]
+        A list of absolute date annotations.
+    """
+    base_date = event_time
+    conf = {"RELATIVE_BASE": base_date}
+    transcribed_dates = []
+    for e in entities:
+        try:
+            parsed_date = dateparser.parse(e.text, settings=conf)
+            transcribed_dates.append(
+                _span_to_annotation(e, "Entity[Date]", parsed_date)
+            )
+        except ValueError:
+            pass
+    return transcribed_dates
+
+
+def _annotate_product_entities(
+    entities: List[spacy.tokens.Span],
+    doc: spacy.tokens.Doc,
+    event_time: datetime
+) -> List[EntityAnnotation]:
+    """
+    Produces annotations of product names, removing commong false positives.
+
+    False positives removed:
+        - "amendment"
+
+    Parameters
+    ----------
+    entities: List[spacy.tokens.Span]
+        A list of spacy.tokens.Span objects representing entities
+    doc: spacy.tokens.Doc
+        The spacy doc the entities are derived from
+    event_time: datetime
+        The timestamp that the event occurred on
+
+    Returns
+    -------
+    annotations: List[EntityAnnotation]
+        A list of product entity annotations
+    """
+    filtered = [e for e in entities if e.text.lower() != "amendment"]
```

Comment:
Could this list of `false_positives` be added as a `kwarg` to the entire `EntityAnalyzer` object. This allows me to include them in the pipeline config instead of in the code so that down the line if a different instance has different specific things to ignore.